virttest/qemu_devices/qdevice_format.py

```diff
@@ -0,0 +1,549 @@
+                "cache-size": self._str_to_dec,
+                "timeout": self._str_to_dec,
+            },
+            "device": {
+                # "general" is NOT a kind of driver.
+                # The arguments in "general" are needed by
+                # at least 2 drivers.
+                "general": {
+                    "write-cache": self._unchanged,
+                    "disable-legacy": self._unchanged,
+                    "serial": self._unchanged,
+                },
+                "virtio-scsi-pci": {
+                    "virtqueue_size": self._str_to_dec,
+                    "num_queues": self._str_to_dec,
+                    "max_sectors": self._str_to_dec,
+                },
+                "virtio-rng-pci": {
+                    "period": self._str_to_dec,
+                    "max-bytes": self._str_to_dec,
+                },
+                "scsi-hd": {
+                    "discard_granularity": self._str_to_dec,
+                    "physical_block_size": self._str_to_dec,
+                    "logical_block_size": self._str_to_dec,
+                    "bootindex": self._str_to_dec,
+                },
+                "virtio-blk-pci": {
+                    "max-write-zeroes-sectors": self._str_to_dec,
+                    "queue-size": self._str_to_dec,
+                    "max-discard-sectors": self._str_to_dec,
+                    "num-queues": self._str_to_dec,
+                    "discard_granularity": self._str_to_dec,
+                },
+                "usb_driver": {
+                    "port": self._str,
+                    "serial": self._on,
+                    # min_io_size, opt_io_size from device ("driver": "usb-storage")
+                    "min_io_size": self._str_to_dec,
+                    "opt_io_size": self._str_to_dec,
+                },
+                "pcie-root-port": {
+                    "port": self._hex_in_str_to_dec,
+                },
+                "nvdimm": {
+                    "label-size": self._normalize_data_size,
+                },
+                "pc-dimm": {
+                    "node": self._str_to_dec,
+                    "addr": self._hex_in_str_to_dec,
+                },
+                "virtio-mem-pci": {
+                    "requested-size": self._normalize_data_size,
+                },
+                "intel-iommu": {
+                    "aw-bits": self._str_to_dec,
+                    "wwn": self._hex_in_str_to_dec,
+                    "eim": self._unchanged,
+                    "intremap": self._unchanged,
+                },
+                "virtio-net-pci": {
+                    "acpi-index": self._str_to_dec,
+                    "host_mtu": self._str_to_dec,
+                    "speed": self._str_to_dec,
+                    "vectors": self._str_to_dec,
+                },
+                "virtio-balloon-ccw": {
+                    "guest-stats-polling-interval": self._str_to_dec,
+                },
```

Comment:
@nickzhq 
I think this needs to be applied as well to virtio-mem-pci since I am hitting the error with it as well

Reply:
Done and thanks @mcasquer